### ISSUE TITLE
Grand Theft Auto Liberty City Stories and Vice City stories transparency bug

### ISSUE THREAD

**Issue by Blackbird88** [PCSX2/pcsx2#378]
I already reported this years ago and it was closed on googlecode as won't fix, but I just want to make sure this bug is known.
LCS
http://i.imgur.com/fYr1u.jpg
VCS
http://i.imgur.com/DeyzT.jpg
Googlecode issue
https://code.google.com/p/pcsx2/issues/detail?id=943

**Comment by gregory38** [comment]
Please retry latest git with
* opengl renderer
* get an openGL 4.5 capable driver
* set m_accurate_blend = 1 in GSdx.ini manually (until someone add an option for it)

**Comment by Blackbird88** [comment]
That depends if GTX 580 is OGL 4.5 capable.

**Comment by mirhl** [comment]
May depend on #600

**Comment by Blackbird88** [comment]
Somehow I doubt that. This is transparency/alpha issue not depth issue.

**Comment by gregory38** [comment]
it could be. It is hard to say.

**Comment by Blackbird88** [comment]
LCS/VCS don't use same engine as III/VC/SA and they don't suffer from any other sever graphical glitches ever since they worked in PCSX2. This bug is only one that persisted.

**Comment by aap** [comment]
I tried to look into that bug a while ago by analyzing some frames with PIX (d3d9 hw) but for some reason PIX couldn't deal with them at all. The draw calls weren't displayed correctly when making a full snapshot (all render targets displayed wrong content) and while recording a stream showed the draw calls correctly I couldn't dissect the meshes and whatnot. I was able to figure out the VCS trails effect with a combination of both and I did see frames that didn't have that problem. These were always the very first frames to be displayed at all but I don't really know what you can conclude from this, maybe I'll take another look. I *think* the pixels failed the depth test but I'm not quite sure. Perhaps in the render target z isn't cleared properly? What you see behind the text is the sky box in any case.

**Comment by Blackbird88** [comment]
Here is the GS Dump you asked for @gregory38 
http://www.mediafire.com/download/a7s064b4roz1cod/VCSAlphaIssue.zip

**Comment by gregory38** [comment]
Did you test the "preload frame" hack?

**Comment by Blackbird88** [comment]
Yes. I turned all the acurrate options and that preload hack.

**Comment by Blackbird88** [comment]
@gregory38 
Not to be pushy or anything, but do you know why this bug exists yet please?

**Comment by gregory38** [comment]
GS supports read/write texture. Standard GPU are either read or write (it is more complex but that the idea). With latest gl extension, we have a limited rw texture. But now texture cache need to be updated (and the gl renderer too)

**Comment by gregory38** [comment]
Well, I know at least a gta game that suffer of this issue. But I didn't check if it is the same issue as this report.

**Comment by Blackbird88** [comment]
Thanks! Hope it'll get fixed eventually.

**Comment by aap** [comment]
It's probably not the same issue I suspect because it looks fine in software mode. The GS read/write that happens in San Andreas doesn't work in software mode either.

**Comment by gregory38** [comment]
Yes you're right. I run the gs dump above but I didn't spot any difference between the SW and the HW renderer. My logging detect some overlapping. But I know that gs dump doesn't support properly depth transfer.
```
Type:Others	ID:3405	Severity:Info	Message:TC: dst Color hit (full): 4 (0x3c00, F:0x0)
Type:Others	ID:3408	Severity:Info	Message:TC: Lookup Target(Color) 1280x1024, hit: 4 (0x3c00, F:0x0)
Type:Error	ID:3411	Severity:Mid	Message:ERROR: Source and Target are the same!
Type:Others	ID:3411	Severity:Info	Message:TC: Remove Src Texture: 483 (0x3c00)
Type:Others	ID:3411	Severity:Info	Message:TC: Lookup Target(Color) 1280x1024, hit: 4 (0x3c00, F:0x0)
Type:Others	ID:3411	Severity:Info	Message:TC: dst Color hit (full): 482 (0x3e00, F:0x0)
Type:Others	ID:3414	Severity:Info	Message:TC: Lookup Target(Color) 1280x1024, hit: 482 (0x3e00, F:0x0)
Type:Error	ID:3417	Severity:Mid	Message:ERROR: Source and Target are the same!
Type:Others	ID:3417	Severity:Info	Message:TC: Remove Src Texture: 483 (0x3e00)
Type:Others	ID:3417	Severity:Info	Message:TC: Lookup Target(Color) 1280x1024, hit: 482 (0x3e00, F:0x0)
Type:Others	ID:3417	Severity:Info	Message:TC: dst Color hit (full): 4 (0x3c00, F:0x0)
Type:Others	ID:3420	Severity:Info	Message:TC: Lookup Target(Color) 1280x1024, hit: 4 (0x3c00, F:0x0)
Type:Error	ID:3423	Severity:Mid	Message:ERROR: Source and Target are the same!

```

**Comment by Blackbird88** [comment]
Looks like enabling HW depth in OGL renderer causes this now
![image](https://cloud.githubusercontent.com/assets/4414625/14044095/9042ae04-f28c-11e5-809d-4f2c8c4638ee.png)

**Comment by gregory38** [comment]
Do you know what is the first version that introduce it? It is recent?

**Comment by Blackbird88** [comment]
Yes it's very recent. Must've been your OGL changes.

**Comment by gregory38** [comment]
Enable HW hack and add this new one on the ini
```
UserHacks_DisableGsMemClear=1
```
Does it fix your issue?

**Comment by Blackbird88** [comment]
Nope. I'm looking through builds now.

**Comment by Blackbird88** [comment]
Found it. It's caused by PCSX2/pcsx2@6002719
With this commit OGL+HW Depth+Higher than native res causes this.

**Comment by gregory38** [comment]
Do you mean that it is working in native resolution?

**Comment by Blackbird88** [comment]
Yes.

**Comment by gregory38** [comment]
I'm puzzled.

**Comment by Blackbird88** [comment]
Puzzled? As in why it's broken?

**Comment by ssakash** [comment]
maybe one alternative would be to disable the no depth lookup optimization on upscaling resolutions ? but honestly I'm not sure why upscaling has problems with it :\

**Comment by gregory38** [comment]
Well, why do you a need depth buffer that is neither read nor written in upscaling resolution? So I likely miss a piece of the equation.

**Comment by Blackbird88** [comment]
Oh I thought my explanation was bad. Nvm :D

**Comment by gregory38** [comment]
Could you generate me a gs dump?

**Comment by Blackbird88** [comment]
Sure
http://www25.zippyshare.com/v/bzuS9IhE/file.html

**Comment by gregory38** [comment]
Damn, I can't reproduce it. Did you enable any hack? Or other option?

**Comment by Blackbird88** [comment]
No hacks. Unless you count HW depth as hack.
Dne 29. 3. 2016 10:00 AM napsal uživatel "Gregory Hainaut" <
notifications@github.com>:

**Comment by gregory38** [comment]
Could you generate me a long dump that start before the issue appear?

**Comment by Blackbird88** [comment]
Long dump?

**Comment by gregory38** [comment]
I mean a multiframe dump. Full detail here: http://forums.pcsx2.net/Thread-How-to-create-a-proper-GS-dump

In short, ctrl+shift+f8, and keep f8 pressed. Be aware file grow very quickly.

**Comment by Blackbird88** [comment]
Okay
http://www110.zippyshare.com/v/2bsNA4zs/file.html

**Comment by gregory38** [comment]
Did you start the dump before the glitch ? Could you post your GSdx.ini file too.

**Comment by Blackbird88** [comment]
There is no "before". It appears right when I start the game.
http://pastebin.com/5cFbMu2s
Happens with clean made INI too.

**Comment by gregory38** [comment]
Euh, I guess you have some logo, start menu etc before, no? Did you use a save state?

**Comment by Blackbird88** [comment]
I did. Does that matter?

**Comment by gregory38** [comment]
That a good question, I don't have the answer.

**Comment by willkuer** [comment]
Still the enulator should work without savestates. Therefore testcase should as well always be without savestates.

**Comment by gregory38** [comment]
Will be fixed once the depth PR is merged.

**Comment by Blackbird88** [comment]
Yeah I will close this when it's merged. Don't worry :)

**Comment by Blackbird88** [comment]
Fixed in PCSX2/pcsx2/@c7f0a85

**Comment by gregory38** [comment]
Thanks you to follow your issue :+1:

**Comment by swift21** [comment]
how did you fix that? can you tell me step by step?

**Comment by Blackbird88** [comment]
There's no step-by-step. Just downloading [latest nightly](http://buildbot.orphis.net/pcsx2/index.php) and selecting OpenGL in GSdx should do the trick.

**Comment by swift21** [comment]
works fine with opengl but opengl also screwed my gpu up. something happens and white-ish colored pixels takes over the display until i change the resolution. is there any solution for this?

**Comment by Blackbird88** [comment]
What is your GPU?

**Comment by swift21** [comment]
hd 7950

**Comment by Blackbird88** [comment]
Try D3D11 then.  If the fix wasn't ported to it then only thing you can try is updating your drivers.

**Comment by swift21** [comment]
ok then, thank you for your help.

**Comment by gregory38** [comment]
AMD drivers got some OpenGL issues. It was reported to them but they didn't manage to fix them. And I'm not even sure that they're still trying...

**Comment by mirh** [comment]
Could you try to put [this dll](http://www81.zippyshare.com/v/WbwRyJys/file.html) in pcsx2 folder, than add `override_GL_ARB_separate_shader_objects=1` to gsdx.ini ?
Make sure blending accuracy is at least basic, or BSOD may happen.

**Comment by swift21** [comment]
it doesnt crash now, but i realised that it is so laggy. i'll stick with d3d11 then i guess, thanks for your help.

**Comment by lightningterror** [comment]
I'll remove this from the 1.6 milestone since the issue is fixed on GL. The fix can be ported on D3D by someone.
